xsuite/src/world/lsworld.ts

```diff
@@ -170,6 +170,47 @@ export class LSWorld extends World {
     return this.proxy.setPreviousBlockInfo(block);
   }
 
+  private async updateCurrentBlockInfo(
+    updateFn: (block: Required<Block>) => Required<Block>,
```

Comment:
1. The output doesn't have to be `Required<Block>`, no? `Block` seems sufficient.

2. If we do this, then I think Block should rather be the version required. And at other places we rather use `Partial<Block>`.

But if abstraction is removed, nothing needs to be addressed here.